File: Easy/[303] Range Sum Query - Immutable/Java Solution/[303] Range Sum Query - Immutable.java
/*
Accepted
303 [Easy]
Runtime: 54 ms, faster than 58.18% of Java online submissions for Range Sum Query - Immutable.
Memory Usage: 16.76 MB, less than 75.11% of Java online submissions for Range Sum Query - Immutable.
*/
class NumArray {
    private int[] nums;
    public NumArray(int[] nums) {
        this.nums = nums;
    }
    
    public int sumRange(int left, int right) {
        int sum = 0;
        for(int i = left; i <= right; i ++)
            sum += this.nums[i];
        return sum;
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */

File: Easy/[303] Range Sum Query - Immutable/Python3 Solution/[303] Range Sum Query - Immutable.py
"""
Accepted
303 [Easy]
Runtime: 2097 ms, faster than 15.99% of Python3 online submissions for Range Sum Query - Immutable.
Memory Usage: 20.80 MB, less than 6.29% of Python3 online submissions for Range Sum Query - Immutable.
"""
class NumArray:

    def __init__(self, nums: List[int]):
        self.nums = nums

    def sumRange(self, left: int, right: int) -> int:
        sum = 0
        for i in range(left, right + 1):
            sum += self.nums[i]
        return sum

# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)